src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -40,7 +42,7 @@ export const PartyMap: React.FC = () => {
     setSelectedRoom(undefined);
   }, []);
 
-  if (!user || !profile) return <>Loading..</>;
+  if (!user || !profile || !isCustomSoundsLoaded) return <>Loading..</>;
```

Comment:
Since we don't actually use the custom sounds directly within this component (but within a subcomponent) it feels weirdly 'disconnected' to have this code here, and feels like it will be quite brittle/easy to miss when making changes due to it being 'spread out' across multiple components.

I think a better way would be to create a new wrapping component for this, probably that works similarly to how `AuthIsLoaded` (in `src/index.tsx`) works. Basically, waits until the sound config is loaded, and then renders the `children`.

This can probably be done directly within the `CustomSoundsProvider` component, though probably should be controlled by a new prop (eg. `waitTillConfigLoaded: boolean`, or something similar but better named)

I expect we probably want to use the `<LoadingPage />` component here. We can probably 'inject' that via another prop, something like: `loadingComponent: ReactNode` or similar.

Then I would expect the 'provider chain' in `src/index.tsx` to end up looking something like:

```jsx
// ..snip..
  <ReactReduxFirebaseProvider {...rrfProps}>
    <AuthIsLoaded>
      <CustomSoundsProvider loadingComponent={LoadingPage} waitTillConfigLoaded>
        <AppRouter />
      </CustomSoundsProvider>
    </AuthIsLoaded>
  </ReactReduxFirebaseProvider>
// ..snip..
```

Reply:
That's looking better than I've expected. Thank you!